src/OCServo.cpp

```diff
@@ -0,0 +1,481 @@
+#include "OCServo.h"
+
+/* CONSTRUCTORS */
+OCServo::OCServo(byte id, SoftwareSerial *serialPort) : id(id), serial(serialPort){
+    this->portType = SERIAL_SOFTWARE;
+}
+
+OCServo::OCServo(byte id, HardwareSerial *serialPort) : id(id), serial(serialPort){
+    this->portType = SERIAL_HARDWARE;
+}
+
+/* PRIVATE METHODS */
+byte OCServo::getChecksum(byte *data, int size) {
+    byte checksum = 0;
+    for (byte i = 2; i < size; i++)
+    {
+        checksum += data[i];
+    }
+    return ~(checksum & 0xFF);
+}
+
+byte OCServo::baudRateToByte(long baudrate) {
+    byte value = 0x00;
+
+    switch (baudrate) {
+        case 500000:
+            value = OCS_BAUDRATE_500K;
+            break;
+        case 250000:
+            value = OCS_BAUDRATE_250K;
+            break;
+        case 128000:
+            value = OCS_BAUDRATE_128K;
+            break;
+        case 115200:
+            value = OCS_BAUDRATE_115200;
+            break;
+        case 76800:
+            value = OCS_BAUDRATE_76800;
+            break;
+        case 57600:
+            value = OCS_BAUDRATE_57600;
+            break;
+        case 38400:
+            value = OCS_BAUDRATE_38400;
+            break;
+        default:
+            // Fallthrough to default 1M
+        case 1000000:
+            value = OCS_BAUDRATE_1M;
+            break;
+    }
+
+    return value;
+}
+
+long OCServo::byteToBaudRate(byte value) {
+    long result;
+
+    switch(value) {
+        case OCS_BAUDRATE_1M:
+            result = 1000000;
+            break;
+        case OCS_BAUDRATE_500K:
+            result = 500000;
+            break;
+        case OCS_BAUDRATE_250K:
+            result = 250000;
+            break;
+        case OCS_BAUDRATE_128K:
+            result = 128000;
+            break;
+        case OCS_BAUDRATE_115200:
+            result = 115200;
+            break;
+        case OCS_BAUDRATE_76800:
+            result = 76800;
+            break;
+        case OCS_BAUDRATE_57600:
+            result = 57600;
+            break;
+        case OCS_BAUDRATE_38400:
+            result = 38400;
+            break;
+        default:
+            result = -1;
+            break;
+    }
+
+    return result;
+}
```

Comment:
Minor: small simplification
```suggestion
uint32_t OCServo::byteToBaudRate(uint8_t value) const {
    switch(value) {
        case OCS_BAUDRATE_1M:      return 1000000;
        case OCS_BAUDRATE_500K:    return 500000;
        case OCS_BAUDRATE_250K:    return 250000;
        case OCS_BAUDRATE_128K:    return 128000;
        case OCS_BAUDRATE_115200:  return 115200;
        case OCS_BAUDRATE_76800:   return 76800;
        case OCS_BAUDRATE_57600:   return 57600;
        case OCS_BAUDRATE_38400:   return 38400;
        default:                   return 0; // Unknown baud rate??? is it possible?
    }
}
```